### PR TITLE
RELATED: RAIL-2787 - Enhance catalog-export

### PR DESCRIPTION
**Pull request by lupko**
This late PR enhances catalog-export to include a new DateDataset mapping. We want to squeeze this into 8.1 because of urgent feedback from services/users/Zajda.

Bottom line:

-  The changes are localized to catalog-export tool and are backward compatible
-  The PR does not touch any of the published libs
-  The PR updates sdk-examples to use the newly generated constants

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [ ] `check-extended` passes
-   [x] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1713/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3291/ : FAILURE in 25m 14s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: tools/catalog-export/src/transform/toTypescript.ts [hunk 1]
```diff
@@ -62,6 +69,37 @@ function uniqueVariable(title: string, nameScope: TakenNamesSet = GlobalNameScop
...
+ * @param nameScope
+ */
+function dateDataSetProperty(title: string, nameScope: TakenNamesSet = GlobalNameScope): string {
+    const datasetStart = title.lastIndexOf("(");
+
+    if (datasetStart === -1) {
+        return uniqueVariable(title, nameScope);
+    }
+
+    return uniqueVariable(title.substr(datasetStart), nameScope);
```
Comment (no23reason) [inline tools/catalog-export/src/transform/toTypescript.ts:90]: What about the closing paren here? I don't see it removed anywhere.

File: tools/catalog-export/src/transform/toTypescript.ts [hunk 2]
```diff
@@ -62,6 +69,37 @@ function uniqueVariable(title: string, nameScope: TakenNamesSet = GlobalNameScop
...
+}
+
+function dateDataSetAttributeProperty(title: string, nameScope: TakenNamesSet = GlobalNameScope): string {
+    const datasetStart = title.lastIndexOf("(");
+
+    if (datasetStart === -1) {
+        return uniqueVariable(title, nameScope);
+    }
+
+    return uniqueVariable(title.substr(0, datasetStart), nameScope);
```
Comment (no23reason) [inline tools/catalog-export/src/transform/toTypescript.ts:100]: What about the closing paren here? I don't see it removed anywhere.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1718/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3296/ : FAILURE in 20m 08s

File: tools/catalog-export/src/transform/toTypescript.ts (hunk 1 above)
Comment (lupko) [reply tools/catalog-export/src/transform/toTypescript.ts:90]: the uniqueVariable takes care of this. its responsibility to make title into a valid and unique variable name.

File: tools/catalog-export/src/transform/toTypescript.ts (hunk 2 above)
Comment (lupko) [reply tools/catalog-export/src/transform/toTypescript.ts:100]: the uniqueVariable takes care of this. its responsibility to make title into a valid and unique variable name.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1721/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3300/ : FAILURE in 13m 59s

**Comment by lupko** [comment]
extended test

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1722/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3301/ : SUCCESS in 24m 16s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/1723/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/534/ : SUCCESS in 32m 11s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/540/ : SUCCESS in 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/800/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/3304/ : SUCCESS in 13m 42s

**Pull request merged**